### PR TITLE
chrisant996.Clink version 1.4.21

### PR DESCRIPTION
**Pull request by chrisant996**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/98160)

### CODE REVIEW

**Comment by chrisant996** [comment]
@microsoft-github-policy-service agree

File: manifests/c/chrisant996/Clink/1.4.21/chrisant996.Clink.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+PackageIdentifier: chrisant996.Clink
+PackageVersion: 1.4.21
+MinimumOSVersion: 10.0.0.0
+Scope: machine
+InstallModes:
+- interactive
+- silent
+UpgradeBehavior: install
+ProductCode: clink_1.4.20.236aba
```
Comment (Trenly) [inline manifests/c/chrisant996/Clink/1.4.21/chrisant996.Clink.installer.yaml:12]:
```suggestion
ProductCode: clink_chrisant996
```

**Review by Trenly — CHANGES_REQUESTED** [review]

File: manifests/c/chrisant996/Clink/1.4.21/chrisant996.Clink.installer.yaml (hunk 1 above)
Comment (chrisant996) [reply manifests/c/chrisant996/Clink/1.4.21/chrisant996.Clink.installer.yaml:12]: Ah, interesting, thanks @Trenly, I didn't realize wingetcreate wouldn't know to update the product code.  Hopefully this is just a one-time change that needs to happen!

**Review by Trenly — APPROVED** [review]

**Comment by chrisant996** [comment]
It says "Installer Validation failed".

But there is no information at all about the failure, or even what it did that resulted in a failure.

So I have no idea how to begin to investigate or fix the unspecified "failure".  😢

Here is the full info from "View more info on Azure Pipelines" link:

```plain
3/2/2023 4:06:29 PM

****************************************************************************************
Installation Verification
****************************************************************************************

-----------------------------------------------------
Failed
-----------------------------------------------------
Status: Completed
```

**Comment by chrisant996** [comment]
@Trenly oh actually I have an idea -- I bet that it uses `clink*setup.exe /S` to attempt an install.  I noticed the other day that the `/S` flag results in a non-zero exit code from the installer, even though the installation succeeds.  Maybe the bot is responding to that?  But then why have all the other manifest updates via bot succeeded?

**Comment by chrisant996** [comment]
@Trenly it's likely to just keep failing. It thinks there's a "security violation", which presumably means malware monitoring got excited. Because Clink inherently has to work by injecting a thread into cmd.exe, that is something that malware suites periodically get overly excited about and mistakenly flag it as malicious. Sometimes releasing a new version changes the bytes just enough that the malware monitoring stops getting tickled, but there's nothing I can directly do to influence that.

**Comment by Trenly** [comment]
> @Trenly it's likely to just keep failing. It thinks there's a "security violation", which presumably means malware monitoring got excited. Because Clink inherently has to work by injecting a thread into cmd.exe, that is something that malware suites periodically get overly excited about and mistakenly flag it as malicious. Sometimes releasing a new version changes the bytes just enough that the malware monitoring stops getting tickled, but there's nothing I can directly do to influence that.

I think it's a bit less finnicky than that, since it passed validation the first time. As Levvie said, the specific error that it's throwing is usually just a temporary error. I honestly wouldn't be surprised if it passed validation this time around, since I've seen back-to-back runs of a PR with the same issue where one failed and the re-run succeeded. We'll just have to wait and see

**Comment by denelon** [comment]
We can also ask @stephengillie to check the logs and possibly do a manual validation if it isn't actually the Malware scans causing the failure.

**Comment by chrisant996** [comment]
If the problem is the /S (silent install) flag returning an exit code of 1, then that's the known issue where NSIS based installers return 1 as the exit code from a successful silent install.  Clink v1.4.22 will include a workaround for that.  But I doubt that's the validation error here, since other versions of Clink were approved and had the same issue.

**Comment by stephengillie** [comment]
Hi @chrisant996,

This application installs normally, but expects arguments when ran:
```
Clink v1.4.21.b59b76
Copyright (c) 2012-2018 Martin Ridgers
Portions Copyright (c) 2020-2023 Christopher Antos
https://github.com/chrisant996/clink

Usage: [options] <verb> [verb_options]

Verbs:
  inject            Injects Clink into a process.
  autorun           Manage Clink's entry in cmd.exe's autorun.
  set               Adjust Clink's settings.
  update            Check for an update for Clink.
  installscripts    Add a path to search for scripts.
  uninstallscripts  Remove a path to search for scripts.
  history           List and operate on the command history.
  info              Prints information about Clink.
  echo              Echo key sequences for use in .inputrc files.
                    ('<verb> --help' for more details)

Options:
  --profile <dir>   Use <dir> as Clink's profile directory.
  --session <id>    Override Clink's session id (for history and info).
  --version         Print Clink's version and exit.
```

**Comment by chrisant996** [comment]
Yes, of course. That's well documented.

However, launching the shortcut that was installed will open a CMD window and inject Clink.

Running clink with no arguments is not a normal usage scenario.

**Comment by stephengillie** [comment]
There wasn't a shortcut installed in my sandbox. I found the executable through the WinGetFileWatcher's output.

**Comment by chrisant996** [comment]
Where did you look for a shortcut? Did you look in the Start Menu? It installs a shortcut in the Clink folder it adds to the current user's Start Menu.

**Comment by stephengillie** [comment]
> @stephengillie Where did you look for a shortcut? There should be one in the Start Menu, in the Clink folder it adds to the current user's Start Menu.

Oh, I didn't think to look there. That works great. 
image

**Comment by chrisant996** [comment]
Excellent 🙂

**Pull request merged**